srtcore/channel.cpp

```diff
@@ -162,10 +162,10 @@ void CChannel::open(const sockaddr* addr)
       if (0 != ::getaddrinfo(NULL, "0", &hints, &res))
          throw CUDTException(MJ_SETUP, MN_NORES, NET_ERROR);
 
-      if (0 != ::bind(m_iSocket, res->ai_addr, res->ai_addrlen))
+      if (0 != ::bind(m_iSocket, res->ai_addr, (int) res->ai_addrlen))
```

Comment:
Here `socklen_t`, too... probably.

Reply:
The last argument of the `::bind` has type `int`. I think casting to `socklen_t` is not correct, although it is a typedef of `int`.

Reply:
Depends. The version that I found in my manpage has the 3rd argument of `::bind` set as `socklen_t`. Might be that there's some difference on Windows, but still the type of this argument should be the same as the type of `addrinfo::ai_addrlen`. This cast is then at least nonportable.

Reply:
@ethouris, you are right on this. `socklen_t` is better.